plugin/group/rollingupdate.go

```diff
@@ -64,20 +66,41 @@ func (r *rollingupdate) waitUntilQuiesced(pollInterval time.Duration, expectedNe
 				return err
 			}
 
+			// The update is only concerned with instances being created in the course of the update.
+			// The health of instances in any other state is irrelevant.  This behavior is important
+			// especially if the previous state of the group is unhealthy and the update is attempting to
+			// restore health.
 			matching, _ := desiredAndUndesiredInstances(instances, r.updatingTo)
 
-			// We are only concerned with the expected number of instances in the desired state.
-			// For example, if the original state of the group was failing to successfully create instances,
-			// old instances may never show up.  We should, however, avoid proceeding if the new instances
-			// are not showing up.
-			if len(matching) >= int(expectedNewInstances) {
+			// Now that we have isolated the instances with the new configuration, check if they are all
+			// healthy.  We do not consider an update successful until the target number of instances are
+			// confirmed healthy.
+			// The following design choices are currently implemented:
+			//
+			//   - the update will continue indefinitely if one or more instances are in the
+			//     flavor.UnknownHealth state.  Operators must stop the update and diagnose the cause.
+			//
+			//   - the update is stopped immediately if any instance enters the flavor.Unhealthy state.
+			//
+			//   - the update will proceed with other instances immediately when the currently-expected
+			//     number of instances are observed in the flavor.Healthy state.
+			//
+			numHealthy := 0
+			for _, inst := range matching {
+				switch r.scaled.Health(inst) {
```

Comment:
This call to get health reaches out to the flavor plugin to get health, which is a RPC call.  How do we handle the cases when this call blocks for a long time?   It would seem that the call would block for a long time especially when often 'unhealthy' means timing out after a long wait.  Also this is iterating over several instances, one by one, I wonder if this will cause problems within the polling loop. 

Maybe N go routines with timeout / cancellation with timeout < pollInterval?